backend/src/memberships/dto/create-membership.dto.ts

```diff
@@ -1,4 +1,17 @@
-import { OmitType } from '@nestjs/swagger';
+import { PickType } from '@nestjs/swagger';
+import { IsDate, IsOptional } from 'class-validator';
 import { MembershipDto } from './membership.dto';
 
-export class CreateMembershipDto extends OmitType(MembershipDto, ['id']) {}
+export class CreateMembershipDto extends PickType(MembershipDto, [
+  'channelId',
+  'userId',
+  'role',
+]) {
+  @IsOptional()
+  @IsDate()
+  mutedUntil?: Date;
+
+  @IsOptional()
+  @IsDate()
+  bannedUntil?: Date;
+}
```

Comment:
This whole file could boil down to:
```typescript
import { PartialType } from '@nestjs/swagger';
import { MembershipDto } from './membership.dto';

export class CreateMembershipDto extends PartialType(MembershipDto) {}
```
It would reduce code duplication. But you would need to add the `@Type(() => Date)` decorator to the properties `mutedUntil` and `bannedUntil` of your `membership.dto.ts`.

Reply:
Wouldn't that make all of the properties `Optional`? The `channelId`, `userId`, and `role` should be mandatory because it wouldn't be possible to make a sensical `Membership` without them.

I agree that there is some code duplication here though. If we can reduce that without sacrificing the mandatory fields, then I have no problem with this approach

Reply:
You are absolutely right!